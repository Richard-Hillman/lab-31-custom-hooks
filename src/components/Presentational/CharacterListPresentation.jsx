import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Character from '../Presentational/Character';
// import CharacterDetailsPresentation from './CharacterDetailsPresentation';

const CharacterListPresentation = ({ characters }) => {
  console.log(characters);
  const characterElements = characters.map(character => (
    <li key={character.name}>
      <Link to={`/characters/${character.name}`}>
        <Character {...character} /> 
      </Link>
    </li>
  ));

  return (
    <ul data-testid="charactersTest">
      {characterElements}
    </ul>
  );
};

CharacterListPresentation.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    image: PropTypes.string,
  }))
};

export default CharacterListPresentation;
