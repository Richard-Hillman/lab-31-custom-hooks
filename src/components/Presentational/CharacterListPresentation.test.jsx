/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterListPresentation from './CharacterListPresentation';

describe('CharacterListPresentation component', () => {
  afterEach(() => cleanup());
  it('renders CharacterListPresentation', () => {
    const { asFragment } = render(<CharacterListPresentation characters={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
