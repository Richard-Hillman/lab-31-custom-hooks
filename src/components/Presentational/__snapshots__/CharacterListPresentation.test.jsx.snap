// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterListPresentation component renders CharacterListPresentation 1`] = `
<DocumentFragment>
  <ul
    data-testid="charactersTest"
  />
</DocumentFragment>
`;
